docs/data/date-pickers/customization-examples/CustomizationExamples.js

```diff
@@ -1,6 +1,70 @@
 import * as React from 'react';
-import CustomizationPlaygroundWrapper from 'docsx/src/modules/components/CustomizationPlaygroundWrapper';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { StaticDatePicker } from '@mui/x-date-pickers/StaticDatePicker';
+import BrandingProvider from 'docs/src/BrandingProvider';
+import CustomizationPlayground from 'docsx/src/modules/components/CustomizationPlayground';
+import CircularProgress from '@mui/material/CircularProgress';
+import config from './examplesConfig';
+
+const useCodeSnippets = (examples) => {
+  const [codeSnippets, setCodeSnippets] = React.useState(null);
+
+  React.useEffect(() => {
+    const fetchCodeSnippets = async () => {
+      const resolvedExamples = { ...examples };
+
+      const promises = Object.entries(examples.examples).reduce(
```

Comment:
> Have you considered loading only the relevant file for a given component and styling option?

I think in that case the `CustomizationPlayground` component would have to worry about loading the code snippet every time the styling option or selected component changes. Which is not bad, I just preferred giving the `CustomizationPlayground` the exact code snippets as a string directly, because that way the component can receive code snippets from anywhere basically. Also, in the future we might want to reuse this demo elsewhere, and load the code examples differently. 
Another thing we could do is have a small context that keeps track of the selected styling option and subcomponent, and that way we can load the relevant code example on a higher level, outside of the `CustomizationPlayground` 🤔 But I'm not sure if I like this option 🙈

Reply:
Oh, I wasn't proposing mixing the data loading logic into the `CustomizationPlayground`, just wanted to say, that aim for simplicity first on the data loading part. 👍